content/guides/upgrading-to-tf12-tg19/index.md

```diff
@@ -0,0 +1,340 @@
+1. [Migrate terraform.tfvars to terragrunt.hcl](#migrate-terraform-tfvars-to-terragrunt-hcl)
+1. [Switch common tfvars files to use yaml](#switch-common-tfvars-files-to-use-yaml)
+
+#### Install and setup Terragrunt 0.19
+
+You will also need to setup the Terragrunt 0.19 binary in a [similar fashion to setting up Terraform
+0.12](#install-and-setup-terraform-0-12). You can use the same instructions available for managing Terraform 0.12 as for
+managing Terragrunt 0.19, except for `tfenv` which only supports Terraform
+
+- [Using homebrew to manage multiple terragrunt versions](#using-homebrew-to-manage-multiple-terraform-versions)
+- [Manually managing multiple terragrunt versions](#manually-managing-multiple-terragrunt-versions)
+
+##### Using homebrew to manage multiple terragrunt versions
+
+Like `terraform`, you can use Homebrew to manage multiple `terraform` versions, using the same method described above in
+[Using homebrew to manage multiple terraform versions](#using-homebrew-to-manage-multiple-terraform-versions). To use
+the method for terragrunt, replace the references for `terraform` with `terragrunt`. For example, in the first step when
+searching for the homebrew commit that introduces the Terragrunt version for `0.18.7`, you would run:
+
+```
+$ git log master -- Formula/terragrunt.rb
+```
+
+instead of the equivalent one for Terraform.
+
+##### Manually managing multiple terragrunt versions
+
+Like `terraform`, you can use the method described above in [Manually managing multiple terraform
+versions](#manually-managing-multiple-terraform-versions) to manage multiple versions of Terragrunt.
+
+You can find the available binaries for each terragrunt version
+[here](https://github.com/gruntwork-io/terragrunt/releases).
+
+
+#### Migrate terraform.tfvars to terragrunt.hcl
+
+Once you have terragrunt 0.19 available on your machine, you will need to migrate your `terraform.tfvars` files to
+`terragrunt.hcl` files before you can start to use the new version. You will need to follow the steps outlined in the
+[Terragrunt 0.19.x migration
+guide](https://github.com/gruntwork-io/terragrunt/blob/master/_docs/migration_guides/upgrading_to_terragrunt_0.19.x.md)
+for each of your live config (e.g in your `infrastructure-live` repo).
+
+You should do this in parallel with each module upgrade. For example, if you were upgrading the module
+`infrastructure-modules/networking/vpc-app`, you should upgrade the live config that deploys that module to a pre-prod
+environment to test the changes you are making to that module (e.g
+[`infrastructure-live/dev/us-east-1/dev/vpc`](https://github.com/gruntwork-io/infrastructure-live-multi-account-acme/tree/dd6dce7f737f8c1bd32466b69e905b2bdd25db80/dev/us-east-1/dev/vpc)).
+The rough process should be:
+
+1. Upgrade the module to TF12 syntax following the steps [listed above](#upgrade-each-module-for-0-12-compatibility).
+1. Push the changes to a branch for testing.
```

Comment:
No commits/pushing for local or pre-prod testing. Instead, use `--terragrunt-source` or `TERRAGRUNT_SOURCE` env var to test with local checkout. Only commit/push/release _after_ local testing, when you are ready to roll the changes to prod envs.

Reply:
Good point. Done!